src/rules/store/prefer-action-creator-in-dispatch.ts

```diff
@@ -33,6 +40,22 @@ export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
       [`${storeDispatch(storeName)} :matches(NewExpression, ObjectExpression)`](
```

Comment:
IMHO, this fix is far from desired, but we can't simply do `${storeDispatch(storeName)} > :matches(NewExpression, ObjectExpression)` (note the `>`) because by doing so, we would miss several possible cases, like:

```ts
this.store$.dispatch(useObject ? { type: 'custom' } : new CustomAction())
this.store$.dispatch(somethingNullable ?? { type: 'custom' })
```

and much more... the expressions aren't necessarily the `first-child`.

---

I am increasingly convinced that the correct approach to getting `constructor` instances is the one I submitted at https://github.com/timdeschryver/eslint-plugin-ngrx/pull/221, which, in addition to looking more performatic, since we would query `store$`/`actions$/`whater` only once and look for its references instead of two (or in this case, **three**), it is also less prone to false positives, as we **do** know that the reference is in fact accurate.